pkg/instrumentation/podmutator.go

```diff
@@ -104,14 +111,25 @@ func (pm *instPodMutator) Mutate(ctx context.Context, ns corev1.Namespace, pod c
 	}
 	insts.DotNet = inst
 
+	if inst, err = pm.getInstrumentationInstance(ctx, ns, pod, annotationInjectGo); err != nil {
+		// we still allow the pod to be created, but we log a message to the operator's logs
+		logger.Error(err, "failed to select an OpenTelemetry Instrumentation instance for this pod")
+		return pod, err
+	}
+	if EnableGoAutoInstrumentationSupport.IsEnabled() {
+		insts.Go = inst
+	} else {
+		logger.Error(err, "support for Go auto instrumentation is not enable")
```

Comment:
```suggestion
		logger.Error(err, "support for Go auto instrumentation is not enabled")
```

Reply:
I think its fine to not return the error here, just expecting someone to come in to the channel and wonder why their go instrumentation isn't working without seeing this log.

Reply:
Absolutely.  Thats gonna be a problem for all the languages once we put them behind feature gates whether or not they are enabled/disabled by default.  I think the change I made in https://github.com/open-telemetry/opentelemetry-operator/pull/1629 to print out the feature gates in the operator's startup will help as well as this log.  But ya, the users is gonna have to go searching for the information on "why it didn't work".  Thats the downside of how the operator manages things right?  It can't give pod owners direct feedback.

Reply:
yeah its a double edged sword... just calling it out as something we should be prepared for :)